### PR TITLE
Add heading component

### PR DESCRIPTION
**Pull request by CJWaszczuk**
# What?
Adds Heading component

# Why?
We need to be able to create headings, without having to directly reference the styles in `lbh-frontend`

# References (Look at the classes for heading)

`lbh-frontend`: [GitHub](https://github.com/LBHackney-IT/LBH-frontend/blob/master/src/lbh/core/_typography.scss) | [Style guide](https://lbh-frontend.herokuapp.com/examples/typography)
`govuk-frontend`: [GitHub](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/core/_typography.scss) | [Style guide](https://design-system.service.gov.uk/styles/typography/)

### CODE REVIEW

File: src/components/typography/Heading.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   */
+  largeLight?: boolean | null;
+  /**
+   * The heading text.
+   */
+  children: React.ReactNode;
+}
+
+/**
+ * A component that creates a heading tag.
```
Comment (erbridge) [inline src/components/typography/Heading.tsx]:
```suggestion
 * A component that creates a heading tag (`<h1>`, `<h2>`, etc.).
```

File: src/components/typography/Heading.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  id?: string | null;
+  className?: string | null;
+  level: HeadingLevels;
+  /**
+   * Style large with a light font.
+   *
+   * See `.lbh-heading-large-light` in `lbh-frontend` for the implementation
+   * details.
+   */
+  largeLight?: boolean | null;
```
Comment (CJWaszczuk) [inline src/components/typography/Heading.tsx]: @LBHELewis Whats the purpose of this class? So we can document it better.

File: src/components/typography/Heading.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+/**
+ * A component that creates a heading tag.
+ */
+export const Heading: React.FunctionComponent<HeadingProps> = (
+  props: HeadingProps
+): JSX.Element => {
+  const { id, level, largeLight, children } = nullValuesAsUndefined(props);
+
+  const HeadingLevel = `h${level}` as "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
```
Comment (erbridge) [inline src/components/typography/Heading.tsx]:
```suggestion
  const H = `h${level}` as "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
```

No need to change it, but I think this is clearer in some ways. It's not the level, it's the tag itself. For future reference.

File: src/components/typography/Heading.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    HeadingLevels.H1,
+    HeadingLevels.H2,
+    HeadingLevels.H3,
+    HeadingLevels.H4,
+    HeadingLevels.H5,
+    HeadingLevels.H6
+  ]).isRequired,
+  largeLight: PropTypes.bool,
+  children: PropTypes.node.isRequired
+} as PropTypes.ValidationMap<HeadingProps>;
```
Comment (erbridge) [inline src/components/typography/Heading.tsx]: 👍

File: src/components/typography/Heading.test.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import { create } from "react-test-renderer";
+
+import { Heading, HeadingLevels } from "./Heading";
+
+it("renders correctly with all props", () => {
+  const component = create(
+    <Heading id="1234" level={HeadingLevels.H1} largeLight={true}>
```
Comment (erbridge) [inline src/components/typography/Heading.test.tsx]: `className` is missing.

File: src/components/typography/Heading.tsx (hunk 2 above)
Comment (LBHELewis) [reply src/components/typography/Heading.tsx]: @CJWaszczuk It's just a lighter-weight heading style defined in our Style Guide. I haven't actually seen it in use anywhere and the reason for its existence hasn't been documented by the designer, so I'm happy for you to omit at this stage if that is easier?

**Review by erbridge — APPROVED** [review]

File: src/components/typography/Heading.tsx (hunk 2 above)
Comment (CJWaszczuk) [reply src/components/typography/Heading.tsx]: Ok, if that's the case, we'll drop it for now. Thanks for the quick response! :)

**Pull request merged**